feat(language_server): switch `tower-lsp` to `tower-lsp-server`

Boshen opened the pull request:
`tower-lsp` has been unmaintained for a longer time. `tower-lsp-server` is a community fork.

Haven't tested with vscode yet.

Boshen commented on crates/oxc_language_server/src/main.rs:
```diff
@@ -83,8 +83,8 @@ impl Options {
 
-#[tower_lsp::async_trait]
 impl LanguageServer for Backend {
+    #[expect(deprecated)] // TODO: FIXME
```
Don't know what this is about 

```
warning: use of deprecated field `tower_lsp_server::lsp_types::InitializeParams::root_uri`: Use `workspace_folders` instead when possible
  --> crates/oxc_language_server/src/main.rs:89:19
   |
89 |         self.init(params.root_uri)?;
```

Sysix replied:
LSP can support multiple workspaces, in VS Code we build the file URI with `workspaceFolders`:
https://github.com/oxc-project/oxc/blob/876b6ea9c033d4845a032327a88b322aa312431e/editors/vscode/client/extension.ts#L305-L309

We should fix this because we are telling the clients that we are supporting it:
https://github.com/oxc-project/oxc/blob/876b6ea9c033d4845a032327a88b322aa312431e/crates/oxc_language_server/src/capabilities.rs#L49-L52

Boshen replied:
Ok, let's do this once this is merged.

Sysix commented:
> Haven't tested with vscode yet.

Commands that shut down the server are failing now.
STRG + P and search for `Oxc: Restart Oxc Server` and `Oxc: toggle enable`

Anything else works 👍 

![grafik](https://github.com/user-attachments/assets/213d6dd4-1a0c-4df1-9bb8-d4e5ee7b5c6f)

````
2025-04-09 18:10:56.499 [info] [Trace - 6:10:56 PM] Sending request 'shutdown - (11)'.
2025-04-09 18:10:56.502 [info] 
thread 'main' panicked at crates/oxc_language_server/src/main.rs:503:43:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
````

![grafik](https://github.com/user-attachments/assets/55cc4195-09d8-46f1-a0c3-91d5080a5c64)

Sysix commented:
### Merge activity

* **Apr 9, 12:58 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Boshen committed:
feat(language_server): switch `tower-lsp` to `tower-lsp-server` (#10298)

`tower-lsp` has been unmaintained for a longer time. `tower-lsp-server` is a community fork.

Haven't tested with vscode yet.
